cupyx/scipy/signal/signaltools.py

```diff
@@ -0,0 +1,81 @@
+import math
+
+import cupy
+
+
+def _numeric_arrays(arrays, kinds='buifc'):
+    if type(arrays) == cupy.ndarray:
+        return arrays.dtype.kind in kinds
+    for array in arrays:
+        if array.dtype.kind not in kinds:
+            return False
+    return True
+
+
+def choose_conv_method(in1, in2, mode='full'):
+    """Find the fastest convolution/correlation method.
+
+    Args:
+        in1 (cupy.ndarray): first input.
+        in2 (cupy.ndarray): second input.
+        mode (str, optional): `valid`, `same`, `full`
+
+    Returns:
+        str: A string indicating which convolution method is fastest,
+         either ‘direct’ or ‘fft’.
+
+    .. warning::
+        This function currently doesn't support measure option,
+        nor multidimensional inputs.
+
+    .. seealso:: :func:`scipy.signal.choose_conv_method`
+
+    """
+    if in1.ndim != 1 or in2.ndim != 1:
+        raise NotImplementedError('Only 1d inputs are supported currently')
+
+    if _numeric_arrays([in1, in2], kinds='bui'):
```

Comment:
`_numeric_arrays` is used only from here. How about embedding directly?
```suggestion
    if in1.dtype.kind in 'bui' or in2.dtype.kind in 'bui':
```